pkg/watch/interface.go

```diff
@@ -0,0 +1,11 @@
+package watch
+
+import (
+	"github.com/redhat-developer/odo/pkg/kclient"
+	"io"
+)
+
+type Client interface {
+	// WatchAndPush watches the component under the context directory and triggers Push if there are any changes
+	WatchAndPush(client kclient.ClientInterface, out io.Writer, parameters WatchParameters) error
```

Comment:
I would prefer to see `client kclient.ClientInterface` injected via Dependency innjection instead of as a parameter here. That would move the dependency to `kclient` from the `dev` package to the `watch` package, what seems more reasonable.

Reply:
Fixed.